modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/multishipping/OrderItemRow.tsx

```diff
@@ -37,6 +40,134 @@ interface IOrderItemRowProps {
 	spritemap?: string;
 }
 
+const calculateOrderItemQuantity = (orderItemDeliveryGroups: {
+	[key: string]: IOrderItemDeliveryGroup;
+}) => {
+	return Object.keys(orderItemDeliveryGroups).reduce(
+		(quantity, deliveryGroupId) => {
+			return quantity + orderItemDeliveryGroups[deliveryGroupId].quantity;
+		},
+		0
+	);
+};
+
+const copyColumnOrderItem = (
+	deliveryGroups: Array<IDeliveryGroup>,
+	orderItem: IOrderItem
+): IOrderItem => {
+	orderItem.deliveryGroups = orderItem.deliveryGroups || {};
```

Comment:
Are all of these impure functions mandatory? You keep rewriting the same memory references, this could become a little dangerous both in terms of control flow and possible debugging. Not a blocker if you're in a rush now, but please keep this in mind.